### ISSUE TITLE
Forecasting-specific algorithms

### ISSUE THREAD

**Issue by James-Large** [sktime/sktime-dl#32]
We now have the reasonable capacity to perform time series forecasting via reduction to regression. Some bespoke forecasting architectures would also be good to implement too, though. 

@mloning this is likely more your ball park to make suggestions for things to be implemented, though I could have a quick read around at some point too

**Comment by mloning** [comment]
Yeah, I'm currently running the regressors on the M4 benchmarking study. 

Regarding forecasting networks, there are some basic architectures: 
* simple RNN used in M4 (see this [file](https://github.com/Mcompetitions/M4-methods/blob/master/ML_benchmarks.py))
* LSTM

Other architectures which would be nice to have, but which I haven't tried myself, are: 

* [ES-RNN](https://github.com/damitkwr/ESRNN-GPU) (M4 winner)
* [N-BEAT](https://github.com/SkanderHn/n-beats-1) (current SOTA afaik)

**Comment by Withington** [comment]
A simple stacked LSTM added as draft PR #43. Here are some other LSTM-based candidates:

### ‘Deep Residual Bidir-LSTM for Human Activity Recognition Using Wearable Sensors’
(Zhao et al., 2017)
https://github.com/guillaume-chevalier  
TensorFlow
“tested with the Opportunity data set and the public domain UCI data set”


### Insights Into LSTM Fully Convolutional Networks for Time Series Classification
(Karim et al., 2019); (Karim et al., 2017)
https://github.com/houshd/LSTM-FCN 
Keras
“Long short-term memory fully convolutional neural networks (LSTM-FCNs) and Attention LSTM-FCN (ALSTM-FCN) have shown to achieve the state-of-the-art performance on the task of classifying time series signals on the old University of California-Riverside (UCR) time series repository. “ (Karim et al., 2019)

### Deep Convolutional and LSTM Recurrent Neural Networks for Multimodal Wearable Activity Recognition
(Ordóñez and Roggen, 2016) https://github.com/sussexwearlab/DeepConvLSTM/blob/master/DeepConvLSTM.ipynb 
Lasagne
“we propose a generic deep framework for activity recognition based on
convolutional and LSTM recurrent units, […] our framework outperforms competing deep non-recurrent networks on the challenge dataset by 4% on average; outperforming some of the previous reported results by up to 9%.”
Datasets: “activity recognition challenge (OPPORTUNITY) and to another open dataset (Skoda)”
